src/Assets/Bugsnag/Editor/BugsnagEditor.cs

```diff
@@ -224,5 +253,125 @@ public static void OnPostProcessBuild(BuildTarget target, string path)
+
+                File.WriteAllText(path, ANDROID_DEPS_XML);
+
+                foreach (var lib in GetKotlinLibs())
+                {
+                    lib.SetCompatibleWithPlatform(BuildTarget.Android, false);
+                    lib.SaveAndReimport();
+                }
+
+                AssetDatabase.Refresh();
+            }
+            catch (Exception e)
+            {
+                Debug.LogError("Error enabling Bugsnag EDM4U support: " + e.Message);
+            }
+
+            if (IsEDMEnabled())
+            {
+                ReportEDMSuccess("Bugsnag EDM4U support successfully enabled.\n\nPlease restart Unity before building.");
+            }
+            else
+            {
+                ReportEDMError("Error enabling Bugsnag EDM4U support.\n\nPlease check the console for error messages");
+            }
+        }
+
+        public static void DisableEDM()
+        {
+            try
+            {
+                var path = Application.dataPath + EDMDepsFilePath;
+                File.Delete(path);
+                File.Delete(path + ".meta");
+
+                foreach (var lib in GetKotlinLibs())
+                {
+                    lib.SetCompatibleWithPlatform(BuildTarget.Android, true);
+                    lib.SaveAndReimport();
+                }
+
+                AssetDatabase.Refresh();
+            }
+            catch (Exception e)
+            {
+                Debug.LogError("Error disabling Bugsnag EDM4U support: " + e.Message);
+            }
+
+            if (!IsEDMEnabled())
+            {
+                ReportEDMSuccess("Bugsnag EDM4U support successfully disabled.\n\nPlease restart Unity before building.");
+            }
+            else
+            {
+                ReportEDMError("Error disabling Bugsnag EDM4U support.\n\nPlease check the console for error messages");
+            }
+        }
+
+        private static void ReportEDMSuccess(string msg)
+        {
+            BugsnagEDMPopup.Open(msg);
+            Debug.Log(msg);
+        }
+
+        private static void ReportEDMError(string msg)
+        {
+            BugsnagEDMPopup.Open(msg);
+            Debug.LogError(msg);
+        }
+
+        private static List<PluginImporter> GetKotlinLibs()
+        {
+            var kotlinLibs = new List<PluginImporter>();
+            foreach (var libPath in Directory.GetFiles(Application.dataPath + KotlinLibsDirPath, "*.jar"))
+            {
+                kotlinLibs.Add((PluginImporter)AssetImporter.GetAtPath(libPath.Replace(Application.dataPath, "Assets")));
+            }
+            return kotlinLibs;
+        }
+
+        private static bool IsEDMEnabled()
+        {
+            var success = File.Exists(Application.dataPath + EDMDepsFilePath);
+            foreach (var lib in GetKotlinLibs())
+            {
+                if (lib.GetCompatibleWithPlatform(BuildTarget.Android))
+                {
+                    success = false;
```

Comment:
You could short-circuit here for a small gain.

Reply:
Done 👍